core/include/list.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+ /**
+ * @addtogroup  core_util
+ * @{
+ *
+ * @file
+ * @brief       Intrusive linked list
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef __LIST_H
+#define __LIST_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct list_node {
+    struct list_node *next;
+} list_node_t;
+
+/**
+ * @brief insert object into list
+ *
+ * @param[in] node      list node before new entry
+ * @param[in] new_node  list node to insert
+ */
+static inline void list_add(list_node_t *node, list_node_t *new_node) {
```

Comment:
I would do

```C
static inline void list_add(list_node_t **node, list_node_t *new_node) {
    if (node == NULL) {
        *node = new_node;
    }
    else {
        new_node->next = node->next;
        node->next = new_node;
    }
}
```

To make the API more consistent with `clist_node_t` (empty list is NULL).